src/main/java/com/google/api/generator/engine/ast/NewObjectExpr.java

```diff
@@ -38,29 +38,24 @@ public static Builder builder() {
         .setIsGeneric(false);
   }
 
-  public static Builder genericBuilder() {
-    return new AutoValue_NewObjectExpr.Builder()
-        .setArguments(Collections.emptyList())
-        .setIsGeneric(true);
-  }
-
   @AutoValue.Builder
   public abstract static class Builder {
     public abstract Builder setType(TypeNode type);
 
     public abstract Builder setArguments(List<Expr> arguments);
 
-    // Private.
-    abstract Builder setIsGeneric(boolean isGeneric);
+    public abstract Builder setIsGeneric(boolean isGeneric);
 
     abstract NewObjectExpr autoBuild();
 
     public NewObjectExpr build() {
       NewObjectExpr newObjectExpr = autoBuild();
-      // Check the object is reference type.
+      // 1. New object expression should be reference type.
       Preconditions.checkState(
-          TypeNode.isReferenceType(newObjectExpr.type()), "New object cannot be reference types.");
-      // Check if there is a conflict between isGeneric() setting and generics() setting.
+          TypeNode.isReferenceType(newObjectExpr.type()),
+          "New object expression should be reference types.");
```

Comment:
I think we could add one more precondition to check if the type is null type. Similar to
```
Preconditions.checkState(
            !newObjectExpr.type().equals(TypeNode.NULL), "New object expression should not be null type");
```

Reply:
I think there's a small chance for users to set the new object expression as null type, but added the check for safety. And also a unit test.